lib/Doctrine/Common/DataFixtures/ReferenceRepository.php

```diff
@@ -101,10 +102,18 @@ protected function getIdentifier($reference, $uow)
     public function setReference($name, $reference)
     {
         $this->references[$name] = $reference;
+
         // in case if reference is set after flush, store its identity
         $uow = $this->manager->getUnitOfWork();
-        if ($uow->isInIdentityMap($reference)) {
-            $this->identities[$name] = $this->getIdentifier($reference, $uow);
+
+        if ($this->manager instanceof PhpcrDocumentManager) {
+            if ($uow->contains($reference)) {
+                $this->identities[$name] = $this->getIdentifier($reference, $uow);
```

Comment:
rather than duplicating all the logic getting the identity, you should change only the detection of the object, not the whole logic